examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/endpoints/MultipartHttpEndpoint.scala

```diff
@@ -0,0 +1,29 @@
+package com.example.http4s.blaze.demo.server.endpoints
+
+import cats.effect.Sync
+import cats.implicits._
+import com.example.http4s.blaze.demo.server.service.FileService
+import org.http4s._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.multipart.Part
+
+class MultipartHttpEndpoint[F[_]](fileService: FileService[F])
+                                 (implicit F: Sync[F]) extends Http4sDsl[F] {
+
+  val service: HttpService[F] = HttpService {
+    case GET -> Root / ApiVersion / "multipart" =>
+      Ok("Send a file (image, sound, etc) via POST Method")
+
+    case req @ POST -> Root / ApiVersion / "multipart" =>
+      req.decodeWith(multipart[F], strict = true) { response =>
+        def filterFileTypes(part: Part[F]): Boolean = {
+          part.headers.exists(_.value.contains("filename"))
```

Comment:
I thought we added a filename helper? if not then we should.

Reply:
Mmm link?